op-batcher/driver.go

```diff
@@ -168,61 +170,94 @@ func (l *BatchSubmitter) Stop() {
 	l.wg.Wait()
 }
 
-// loadBlocksIntoState loads all blocks since the previous submitted block
+// loadBlocksIntoState loads all blocks since the previous stored block
+// It does the following:
+// 1. Fetch the sync status of the sequencer
+// 2. Check if the sync status is valid or if we are all the way up to date
+// 3. Check if it needs to initialize state OR it is lagging (todo: lagging just means race condition?)
+// 4. Load all new blocks into the local state.
 func (l *BatchSubmitter) loadBlocksIntoState(ctx context.Context) {
-	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
-	defer cancel()
-	syncStatus, err := l.cfg.RollupNode.SyncStatus(ctx)
+	start, end, err := l.calculateL2BlockRangeToStore(ctx)
 	if err != nil {
-		l.log.Warn("issue fetching L2 head", "err", err)
+		l.log.Trace("was not able to calculate L2 block range", "err", err)
 		return
 	}
-	if syncStatus.HeadL1 == (eth.L1BlockRef{}) {
-		l.log.Info("Rollup node has no L1 head info yet")
-		return
+
+	// Add all blocks to "state"
+	for i := start.Number + 1; i < end.Number+1; i++ {
+		id, err := l.loadBlockIntoState(ctx, i)
+		if errors.Is(err, ErrReorg) {
+			l.log.Warn("Found L2 reorg", "block_number", i)
+			l.state.Clear()
+			l.lastStoredBlock = eth.BlockID{}
+			return
+		} else if err != nil {
+			l.log.Warn("failed to load block into state", "err", err)
```

Comment:
I think this error case will cause it to hit the reorg path the next time and then the state will be cleared and then it'll be able to progress. This happens if the state is partially populated and then `calculateL2BlockRangeToStore` returns a `start` value earlier than what has already been populated